std/math.d

```diff
@@ -4175,6 +4176,41 @@ real round(real x) @trusted nothrow @nogc
         return core.stdc.math.roundl(x);
 }
 
+///
+@safe nothrow @nogc unittest
+{
+    assert(round(0.49) == 0);
+    assert(round(0.5 ) == 0);
+    assert(round(0.51) == 1);
+    assert(round(1.5 ) == 2);
+}
+
+/*******************************************
+ * Return the value of x rounded to the given precision (number of decimal places).
+ *
+ * If the fractional part of x $(I below) the last required decimal place is
+ * exactly 0.5, the last decimal place is rounded to the nearby even digit.
+ *
+ * The precision can be negative, in which case rounding to the nearest multiple
+ * of positive powers of ten (10, 100 etc) will occur.
+ */
+real round(real val, int prec) @safe nothrow @nogc
+{
+    real pow = 10 ^^ prec;
```

Comment:
This will easily and silently overflow, because you are computing the power with `int` arithmetic.

You need to make sure that `10` is already of type `real` *before* being fed into the power operator `^^` to avoid this. Because of [DMD issue #15288](https://issues.dlang.org/show_bug.cgi?id=15288), the obvious solution `cast(real)10 ^^ prec` won't do what you need.

The best way is this: `10.0L ^^ prec`. Adding `.0L` makes `10` into a type `real` literal, instead of the default `int`.